### PR TITLE
feat(status-badge): add borderless size

### PR DESCRIPTION
**Pull request by nkrantz**
Add `size="borderless"` to Status Badge. Update docs and stories.

### CODE REVIEW

File: .changeset/clean-llamas-joke.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@twilio-paste/status": minor
+"@twilio-paste/core": minor
+---
+
+[Status Badge] add new `size` value "borderless" which removes padding and border (box shadow) for a less stylized option of Status Badge for use in Tables and other crowded layouts.
```
Comment (krisantrobus) [inline .changeset/clean-llamas-joke.md:6]: praise: great description

File: packages/paste-core/components/status/src/StatusBadge.tsx [hunk 2]
```diff
@@ -21,10 +22,43 @@ export type StatusBadgeProps = Omit<BadgeBaseProps, "variant"> &
      * @type {StatusBadgeVariants}
      */
     variant: StatusBadgeVariants;
+    /**
+     * Sets the size of the Status Badge
+     *
+     * @default "default"
+     * @type {StatusBadgeVariants}
```
Comment (krisantrobus) [inline packages/paste-core/components/status/src/StatusBadge.tsx:29]: incorrect type

Comment (nkrantz) [reply packages/paste-core/components/status/src/StatusBadge.tsx:29]: good catch! c/p error

**Review by krisantrobus — APPROVED** [review]
LGTM!

**Review by serifluous — APPROVED** [review]

**Pull request merged**